### PR TITLE
hack: customize script for OS type

### PR DESCRIPTION
**Pull request by raghavendra-talur**
The prerequisites are different on MacOS than on Linux. Customize the script accordingly.

### CODE REVIEW

**Review by asn1809 — APPROVED** [review]
lgtm

File: hack/check-drenv-prereqs.sh [hunk 1]
```diff
@@ -16,9 +23,23 @@ for cmd in "${commands[@]}"; do
...
+
+if [[ "$os" == "Linux" ]]
+then
+  for cmd in "${linux_only_commands[@]}"; do
+    if ! command -v "$cmd" &> /dev/null; then
+      echo "Error: $cmd could not be found in PATH."
+      exit 1
+    fi
+  done
+fi
```
Comment (nirs) [inline hack/check-drenv-prereqs.sh:35]: This is unneeded duplication. We could do this instead:

    if linux:
        append virt-host-validate to commands

And use single loop for checking if commands exist.

**Review by nirs — COMMENTED** [review]
These bash scripts become more complicated and using bash cryptic syntax make it harder to maintain. This should move into drenv since it is about checking tools needed by drenv.

We already have a "drenv setup" that can be used to check or better deploy the required tools.

**Comment by raghavendra-talur** [comment]
@nirs I will close this PR and let you handle the requirements in the drenv setup code.

**Pull request closed without merging**